docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,876 @@
+
+Route Processors
+    RP0                 8102-28FH-DPU-O 0.10                 FLM274802F3     Cisco 28x400G QSFPDD DPU-Enabled 2RU Smart Switch,Open SW
+
+Sled Cards
+    SLED0               8K-DPU400-2A    0.10                 FLM2750036M     Cisco 800 2xDPU Sled AMD Elba
+    SLED1               8K-DPU400-2A    0.10                 FLM2750037E     Cisco 800 2xDPU Sled AMD Elba
+    SLED2               8K-DPU400-2A    0.10                 FLM27500389     Cisco 800 2xDPU Sled AMD Elba
+    SLED3               8K-DPU400-2A    0.10                 FLM2750038M     Cisco 800 2xDPU Sled AMD Elba
+
+Dpu Modules
+    DPU0                DSS-MTFUJI      6.1.0-11-2-arm64     FLM2750036M     Pensando DSC
+    DPU1                DSS-MTFUJI      6.1.0-11-2-arm64     FLM2750037M     Pensando DSC
+    DPU2                DSS-MTFUJI      6.1.0-11-2-arm64     FLM2750037E     Pensando DSC
+    DPU3                DSS-MTFUJI      6.1.0-11-2-arm64     FLM2750038E     Pensando DSC
+    DPU4                DSS-MTFUJI                                           Powered off
+    DPU5                DSS-MTFUJI      6.1.0-11-2-arm64     FLM27500390     Pensando DSC
+    DPU6                DSS-MTFUJI      6.1.0-11-2-arm64     FLM2750038M     Pensando DSC
+    DPU7                DSS-MTFUJI      6.1.0-11-2-arm64     FLM2750039M     Pensando DSC
+
+Power Supplies
+    psutray                                                                  
+        PSU0            UCSC-PSU1-2300W A0                   DTM274202UB     UCS 230000W AC-DC High Line RSP02 Power Supply
+        PSU1 -- not present
+
+Cooling Devices
+    fantray0            FAN-2RU-PI-V3   N/A                  N/A             Cisco 8000 Series 2RU Fan with Port-side Air Intake Ver 3
+    fantray1            FAN-2RU-PI-V3   N/A                  N/A             Cisco 8000 Series 2RU Fan with Port-side Air Intake Ver 3
+    fantray2            FAN-2RU-PI-V3   N/A                  N/A             Cisco 8000 Series 2RU Fan with Port-side Air Intake Ver 3
+    fantray3            FAN-2RU-PI-V3   N/A                  N/A             Cisco 8000 Series 2RU Fan with Port-side Air Intake Ver 3
+
+FPDs
+    RP0/info.0                          0.8.0-287                            \_SB_.PC00.RP07.PXSX.INFO
+    RP0/info.1                          0.4.7-122                            \_SB_.PC00.RP01.PXSX.INFO
+    RP0/info.2                          0.2.1-247                            \_SB_.PC00.RP10.PXSX.INFO
+    RP0/info.50.auto                    10.2.0-30                            \_SB_.PC00.RP07.PXSX.P2PF
+
+root@sonic:/home/cisco# config chassis modules startup DPU4
+
+```
+#### Pass/Fail Criteria
+ * Verify dpu powered off in platform inventory after dpu shut down
+ * Verify dpu is shown in platform inventory after dpu powered on
+
+
+### 1.7 Check removal of pcie link between npu and dpu
+
+#### Steps
+ * Use command `pcieutil generate` to generate pcie yaml
```

Comment:
Why pcie.yaml file should be generated in the run-time?

Reply:
It should be part of the platform definition

Reply:
There are use cases where user can turn off and turn on DPUs dynamically using config cli, so in such cases pcieyaml needs to be regenerated.

Reply:
This is not the right approach to do this. The HLD should be extended to cover this case during the shutdown of the DPU. @prgeor what do you think?

Reply:
This  approach is being used as part of log analyzer which is executed as a pretest in most of the sonic-mgmt test. Going forward for smartswitch, the number of dpus and hence number of pcie devices may not remain constant and has to be commuted at the beginning of every test. So, this approach works reliably for all the cases.

Reply:
The removal of the PCI device will be handled in PCIEd as a part of the changes proposed in the Smart Switch reboot HLD (https://github.com/sonic-net/SONiC/pull/1699).

Please remove this test as the suggested flow is not correct.